Kernel/Net/TCPSocket.h

```diff
@@ -129,7 +129,7 @@ class TCPSocket final : public IPv4Socket {
     u32 bytes_out() const { return m_bytes_out; }
 
     KResult send_tcp_packet(u16 flags, const UserOrKernelBuffer* = nullptr, size_t = 0);
-    void send_outgoing_packets();
+    void send_outgoing_packets(RoutingDecision);
```

Comment:
Can we pass this by const-reference instead?

Reply:
I've changed it to `RoutingDecision&`. We can't make this `const` here, because `decision.adapter->send_ipv4()` mutates the adapter and `const RefPtr<NetworkAdapter>` only gives us a `NetworkAdapter const&` for `operator T*`. Not sure that makes sense. Maybe `const RefPtr` should only make the pointer itself const?